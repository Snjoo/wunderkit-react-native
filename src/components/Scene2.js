import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  NavigatorIOS,
  Image
} from 'react-native';

export default class Scene2 extends Component {

  constructor() {
    super();
    this.state = {
      backgroundColor: '#DEA'
    }
  }

  toggleBackground() {
    if (this.state.backgroundColor === '#DEA') {
        this.setState({
          backgroundColor: 'rgba(250, 220, 166, 0.8)'
        });
    } else {
      this.setState({
        backgroundColor: '#DEA'
      })
    }
  }

  render() {
    return (
      <View style={[styles.container, { backgroundColor: this.state.backgroundColor }]}>
        <View style={styles.welcomeContainer}>
          <Text style={styles.welcome}>
            Welcome to Wunderkit!

            Bier ist gut!
          </Text>
        </View>
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            source={{uri: 'https://c1.staticflickr.com/5/4009/4523144699_8bb453698f_m.jpg'}}
            />
        </View>
        <View style={styles.textContainer}>
          <Text style={styles.instructions}>
            To get started, edit index.ios.js
          </Text>
          <Text style={styles.instructions}>
            Press Cmd+R to reload,{'\n'}
            Cmd+D or shake for dev menu
          </Text>
        </View>
        <View style={styles.buttonContainer}>
          <TouchableHighlight underlayColor='red' onPress={() => this.toggleBackground()} style={styles.button}>
            <Text>Switchhh</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  welcomeContainer: {
    flex: 1,
    justifyContent: 'flex-start'
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#0a0a0a'
  },
  textContainer: {
    flex: 1
  },
  imageContainer: {
    padding: 10
  },
  image: {
    width: 150,
    height: 150,
    resizeMode: 'contain'
  },
  button: {
    flex: 1,
    padding: 5,
    margin: 10,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'flex-end',
    backgroundColor: 'orange',
    borderRadius: 3
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
    color: 'white'
  },
  instructions: {
    color: 'white',
    textAlign: 'center',
    marginBottom: 5,
  }
});
