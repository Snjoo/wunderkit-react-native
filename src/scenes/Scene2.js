import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  NavigatorIOS,
  Image
} from 'react-native';

export default class Scene2 extends Component {
  constructor() {
    super()
    this.state = {
      backgroundColor: '#c8bfbf'
    }
  }

  toggleBackgroundColor() {
    if (this.state.backgroundColor === '#c8bfbf') {
      this.setState({backgroundColor: '#6b6b47'})
    } else {
      this.setState({backgroundColor: '#c8bfbf'})
    }
  }

  render() {
    return (
      <View style={[styles.container, {backgroundColor: this.state.backgroundColor}]}>
        <View style={styles.welcomeContainer}>
          <Text style={styles.welcome}>
            This is second scene
          </Text>
          <Image
            source={{uri: 'https://raw.githubusercontent.com/wiki/facebook/react/react-logo-1000-transparent.png'}}
            style={styles.image}
          />
        </View>
        <TouchableHighlight style={styles.button} onPress={() => this.toggleBackgroundColor()}>
          <Text>Toggle color</Text>
        </TouchableHighlight>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  welcomeContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructionsContainer: {
    flex: 1
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  button: {
    height: 40,
    borderWidth: 1,
    borderColor: '#000000',
    backgroundColor: '#a68888',
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch'
  },
  image: {
    resizeMode: 'contain',
    height: 150,
  }
});
