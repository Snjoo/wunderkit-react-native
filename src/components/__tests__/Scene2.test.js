import 'react-native';
import React from 'react';
import Scene2 from '../Scene2';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Scene2 />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
