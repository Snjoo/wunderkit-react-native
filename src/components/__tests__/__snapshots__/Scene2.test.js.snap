// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#0a0a0a",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#DEA",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to Wunderkit! Bier ist gut!
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://c1.staticflickr.com/5/4009/4523144699_8bb453698f_m.jpg",
        }
      }
      style={
        Object {
          "height": 150,
          "resizeMode": "contain",
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "marginBottom": 5,
          "textAlign": "center",
        }
      }
    >
      To get started, edit index.ios.js
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "marginBottom": 5,
          "textAlign": "center",
        }
      }
    >
      Press Cmd+R to reload,
      
      
      Cmd+D or shake for dev menu
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "orange",
            "borderRadius": 3,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "padding": 5,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Switch
      </Text>
    </View>
  </View>
</View>
`;
